packages/block-editor/src/components/button-block-appender/index.js

```diff
@@ -21,17 +21,31 @@ function ButtonBlockAppender( { rootClientId, className } ) {
 			<BlockDropZone rootClientId={ rootClientId } />
 			<Inserter
 				rootClientId={ rootClientId }
-				renderToggle={ ( { onToggle, disabled, isOpen } ) => (
-					<Button
-						className={ classnames( className, 'block-editor-button-block-appender' ) }
-						onClick={ onToggle }
-						aria-expanded={ isOpen }
-						disabled={ disabled }
-					>
-						<span className="screen-reader-text">{ __( 'Add Block' ) }</span>
-						<Icon icon="insert" />
-					</Button>
-				) }
+				renderToggle={ ( { onToggle, disabled, isOpen, blockTitle, hasSingleBlockType } ) => {
+					let label;
+					if ( hasSingleBlockType ) {
+						// translators: %s: the name of the block when there is only one
+						label = sprintf( _x( 'Add %s', 'directly add the only allowed block' ), blockTitle );
+					} else {
+						label = _x( 'Add block', 'Generic label for block inserter button' );
+					}
+					const isToggleButton = ! hasSingleBlockType;
+					return (
+						<Tooltip text={ label }>
+							<Button
+								className={ classnames( className, 'block-editor-button-block-appender' ) }
+								onClick={ onToggle }
+								aria-haspopup={ isToggleButton ? 'true' : undefined }
+								aria-expanded={ isToggleButton ? isOpen : undefined }
+								disabled={ disabled }
+								label={ label }
+							>
+								<span className="screen-reader-text">{ label }</span>
+								<Icon icon="insert" />
+							</Button>
+						</Tooltip>
+					);
```

Comment:
Hmm, this button with an icon and a tooltip is now looking exactly like an `IconButton` 😄 
https://github.com/WordPress/gutenberg/blob/master/packages/components/src/icon-button/index.js